### PR TITLE
Integrate Ballerina services with the WSO2 APIM Service catalog

### PR DESCRIPTION
**Pull request by SasinduDilshara**
Fixes - https://github.com/ballerina-platform/ballerina-lang/issues/42420

Design document - https://docs.google.com/document/d/1oLQLjXm2Uo1YJtqUO3mVnt_tPcRbrFJKloW7Jg3JrcU/edit?usp=sharing

### CODE REVIEW

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java [hunk 1]
```diff
@@ -155,6 +155,14 @@ public BuildCommand() {
     @CommandLine.Option(names = "--cloud", description = "Enable cloud artifact generation")
     private String cloud;
 
+    @CommandLine.Option(names = "--runtime-management-included", description = "package service management tools in " +
+            "the executable JAR file(s).")
+    private Boolean runtimeManagementIncluded;
+
+    @CommandLine.Option(names = "--enable-service-publish", description = "package service management tools " +
+            "for WSO2 API manager in the executable JAR file(s).")
```
Comment (hasithaa) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java:163]: This needs to be updated. This has to give the message "Include ballerina service management tools for publishing service data to external service catalogs."

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-build.help [hunk 2]
```diff
@@ -59,6 +59,14 @@ OPTIONS
        --graalvm-build-options
            Additional build options to be passed to the GraalVM native image.
 
+       --runtime-management-included
+           Include the dependencies that are required to enable runtime package
+           management service.
+
+       --enable-service-publish
+          Include the dependencies that are required to enable runtime package
+          management service for WSO2 API manager.
```
Comment (hasithaa) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-build.help:68]: I would prefer to make this a generic feature, where WSO2 APIM is one of the service catalogs we support.

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java (hunk 1 above)
Comment (warunalakshitha) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java:163]: Can we have one flag for both? Runtime config will decide different options.
ex. enableRemoteManagement

Comment (hasithaa) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java:163]: No, we can't, with this option, we need to enable OAS generation for http services. For other options, this step is not required.

**Comment by SasinduDilshara** [comment]
> @SasinduDilshara, we need to enable OAS generation as part of this. Do we support this?

Initially plan to add a separate PR, I will add it into this PR after having discussion with @lnash94

**Comment by SasinduDilshara** [comment]
> > @SasinduDilshara, we need to enable OAS generation as part of this. Do we support this?
> 
> Initially plan to add a separate PR, I will add it into this PR after having discussion with @lnash94

Had a discussion with @hasithaa @lnash94 @TharmiganK @gayaldassanayake @ShammiL @Thevakumar-Luheerathan and came to the conclusion

Will update the PR

**Comment by isudana** [comment]
Is this  Closed without merging?

**Pull request closed without merging**